Orange/data/table.py

```diff
@@ -126,21 +126,23 @@ def set_weights(self, weight):
             but only if those values are all numbers and are not NA/NaN.
         If a sequence of (non-NA/NaN) numbers, set those values as the sequence.
         """
-        # TODO: handle NAs
         if isinstance(weight, Number):
-            self[Table._WEIGHTS_COLUMN] = weight
+            self[Table._WEIGHTS_COLUMN] = weight if not np.isnan(weight) else 1
         elif isinstance(weight, str):
             if weight not in self.columns:
                 raise ValueError("{} is not a column.".format(weight))
             if self[weight].isnull().any() and np.issubdtype(self[weight].dtype, Number):
                 raise ValueError("All values in the target column must be valid numbers.")
-            self[Table._WEIGHTS_COLUMN] = self[weight]
+            self[Table._WEIGHTS_COLUMN] = self[weight].fillna(value=self[weight].median())
         elif isinstance(weight, (Sequence, np.ndarray)):  # np.ndarray is not a Sequence
             if len(weight) != len(self):
                 raise ValueError("The sequence has length {}, expected length {}.".format(len(weight), len(self)))
             self[Table._WEIGHTS_COLUMN] = weight
+        elif isinstance(weight, pd.Series):
+            # drop everything but the values to uncomplicate things
+            self[Table._WEIGHTS_COLUMN] = list(weight.fillna(value=weight.median()))
```

Comment:
Because
```
df = pd.DataFrame([1, 2, 3], index=[0, 1, 2])
df["newcol"] = pd.Series([4, 5, 6], index=[4, 5, 6])
assert df["newcol"].isnull().all()
```
And setting a list just requires lengths to be the same.